src/_macosx.m

```diff
@@ -1236,7 +1236,7 @@ -(void)drawRect:(NSRect)rect
     CGContextRef cr = [[NSGraphicsContext currentContext] CGContext];
 
     if (!(renderer = PyObject_CallMethod(canvas, "get_renderer", ""))
-        || !(renderer_buffer = PyObject_GetAttrString(renderer, "_renderer"))) {
+        || !(renderer_buffer = PyObject_CallMethod(renderer, "buffer_rgba", ""))) {
```

Comment:
Looking at the implementation of `RendererAgg`, it's:
```
    def buffer_rgba(self):
        return memoryview(self._renderer)
```
so this looks like a reasonable change. I'm a bit uncertain why `PyObject_GetBuffer` is not sufficient with pybind11's version of buffers, but we can investigate that for later.